crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -100,6 +100,17 @@ def knife_files
           filecontent_replace(file, "nova_dashboard", "horizon")
         end
       end
+
+      # find admin node and update target_platform
+      nodes_path = knife_path.join("nodes")
+      nodes_path.children.each do |file|
+        json = JSON.load(file.read)
+        next unless json["crowbar"] && json["crowbar"]["admin_node"]
+        json["target_platform"] = "suse-12.1"
```

Comment:
Yeah, running ohai is a bit overkill. So what about my earlier suggestion: simply `json.delete("target_platform")` and let the provisioner cookbook fill that again?